### PR TITLE
Fixed issue: unresponsiveness to state change

### PR DESCRIPTION
**Pull request by thearthurdev**
I have fixed the issue I reported 4 days ago about the unresponsiveness of the carousel options to state changes.

The majority of the changes in the file is code formatting, sorry about that. 

Please review and merge.

Thanks!

P.S. This is my first open source PR!

### CODE REVIEW

File: lib/carousel_slider.dart [hunk 1]
```diff
@@ -73,97 +76,121 @@ class CarouselSliderState extends State<CarouselSlider> with TickerProviderState
...
+
+    // pagecontroller needed to be re-initialized to respond to state changes
+    // I'm not too sure how bad this is for performance but it works :D
+    pageController = PageController(
+      viewportFraction: options.viewportFraction,
+      initialPage: carouselState.realPage,
+    );
+
+    carouselState.pageController = pageController;
+
```
Comment (thearthurdev) [inline lib/carousel_slider.dart:91]: Calling `handleAutoPlay` and re-initializing the `pageController` to respond to changes in state.

**Comment by serenader2014** [comment]
Hi @thearthurdev , sorry for the late respond, I've been pretty busy these days.

Can you please don't change the code formatting? It makes me hard to review.

**Comment by thearthurdev** [comment]
Hello @serenader2014, sorry about the formatting, It does so automatically in my editor.

**Comment by thearthurdev** [comment]
@serenader2014 I'd like to close this PR and reopen another one where with no changes in the formatting.

**Comment by mwawrusch** [comment]
Any chance to get that pull request and updated? It might solve some other critical error (the one with round on null)

**Comment by thearthurdev** [comment]
I'm closing this PR as it has become unneccessarily hairy. I'll reopen another one with only the neccessary changes.

**Comment by thearthurdev** [comment]
@mwawrusch @serenader2014 New PR can be found here #174

**Pull request closed without merging**